### PR TITLE
post: ng2 - types

### PR DESCRIPTION
**Pull request by cexbrayat**

### CODE REVIEW

File: _posts/2015-10-27-types.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+This dynamic nature allows wonderful things but it is also a pain for a few others compared to more statically typed languages. The most obvious might be when you call an unknown function in JS from an other API, you pretty much have to read the doc (or worse the function code) to know what the parameter should look like. Look at our previous example, the method `printColor` needs a parameter with a `color` property. That can be hard to guess, and of course it is much worse in day to day work, where we use various libraries and services developed by fellow developers. One of Ninja Squad's co-founders is often complaining about the lack of types in JS, and says he can't be as productive and produce as good code as he would in a more statically typed environment. And he is not entirely wrong, even if he is sometimes ranting for sheer pleasure too! Without type information, IDEs have no real clue if you're doing something wrong, and tools can't help you find bugs in your code. Of course, we have tests in our apps, and Angular has always been keen on making testing easy, but it's nearly impossible to have a perfect test coverage.
```
Comment (clacote) [inline _posts/2015-10-27-types.md]: That's a long chapter. Maybe you should introduce linebreaks, before "One of Ninja Squad's cofounder"?

File: _posts/2015-10-27-types.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        this.raceService.list()
+          .then(races => this.races = races);
+        }
+    }
+
+Now the injection can be done! You don't have to use TypeScript in Angular 2, but clearly part of your code will be more elegant if you do. You can always do the same thing in plain ES6 or ES5, but you will have to manually add the metadata in another way (we'll come back on this in more details).
+
+That's why we're going to spend a few time learning TypeScript (TS) in the following posts. Angular 2 is clearly built to leverage ES6 and TS 1.5+, so we will have the easiest time writing our apps using it. And the Angular team really hopes to submit the type system to the standard committee, so maybe one day we'll have types in JS, and all this will be usual.
+
+Stay tune for the next episode on TypeScript features!
```
Comment (clacote) [inline _posts/2015-10-27-types.md]: Maybe add a mention and a link to our ebook, like "and register to our mailing-list if you want to be informed on our ebook progress".

**Comment by clacote** [comment]
Fixed some code spacing in #155.
Otherwise: :+1:

**Comment by cexbrayat** [comment]
Ready to go!

**Comment by clacote** [comment]
https://twitter.com/ninjasquad/status/661445266772525056

**Pull request merged**